netbox/utilities/utils.py

```diff
@@ -174,3 +178,39 @@ def to_meters(length, unit):
     if unit == LENGTH_UNIT_INCH:
         return length * 0.3048 * 12
     raise ValueError("Unknown unit {}. Must be 'm', 'cm', 'ft', or 'in'.".format(unit))
+
+
+@contextmanager
+def pg_advisory_lock(string_key):
+    """
+    Acquire an exclusive database lock against an arbitrary string key
+    """
+    try:
+        lock_key = _get_lock_key(string_key)
+        cursor = connection.cursor()
+        cursor.execute(f'SELECT pg_advisory_lock({lock_key});')
+        yield
+    finally:
+        cursor.close()
+
+
+def with_pg_advisory_lock(string_key):
+    """
+    Decorator that wraps a function with pg_advisory_lock
```

Comment:
Please provide more documentation here as to why this is needed. I would also like to see the link the Postgres docs here too.

Reply:
Done